github.go

```diff
@@ -0,0 +1,145 @@
+package watchdogs
+
+import (
+	"bytes"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/sync/errgroup"
+)
+
+var _ = github.ScopeAdminOrg
+
+var _ CommentService = &GitHubPullRequest{}
+var _ DiffService = &GitHubPullRequest{}
+
+// `path` to `position`(LnumDiff) to comment `body`s
+type postedcomments map[string]map[int][]string
+
+// IsPosted returns true if a given comment has been posted in GitHub already,
+// otherwise returns false. It sees comments with same path, same position,
+// and same body as same comments.
+func (p postedcomments) IsPosted(c *Comment) bool {
+	if _, ok := p[c.Path]; !ok {
+		return false
+	}
+	bodys, ok := p[c.Path][c.LnumDiff]
+	if !ok {
+		return false
+	}
+	for _, body := range bodys {
+		if body == c.Body {
+			return true
+		}
+	}
+	return false
+}
+
+// https://developer.github.com/v3/pulls/comments/#create-a-comment
```

Comment:
comment on exported type GitHubPullRequest should be of the form "GitHubPullRequest ..." (with optional leading article)